doc/guide/Makefile-guide.am

```diff
@@ -98,6 +98,11 @@ dist/guide/html/index.html: $(GUIDE_DOCBOOK) $(GUIDE_INCLUDES) $(man_MANS) $(GUI
 		LANG=en_US.UTF-8 $(XMLTO) html -m $(srcdir)/doc/guide/gtk-doc.xsl -o dist/guide/html/ \
 			--searchpath $(abs_builddir):$(abs_srcdir):$(abs_builddir)/doc/guide \
 			$(srcdir)/$(GUIDE_DOCBOOK)
+		if grep -n -r 'name=\"id' dist/guide/html; then \
+			echo "Unexpected generated id in the documentation"; \
```

Comment:
I'd send the error message and grep output to stderr (`... >&2`) in case this is being run in quiet mode. We actually do this in our CI system (see my recent lintian fix), so this would swallow the error message.